src/torchcodec/decoders/_core/CMakeLists.txt

```diff
@@ -7,58 +7,119 @@ find_package(Torch REQUIRED)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic -Werror ${TORCH_CXX_FLAGS}")
 find_package(Python3 ${PYTHON_VERSION} EXACT COMPONENTS Development)
 
-function(make_torchcodec_library library_name ffmpeg_target)
-    set(
-        sources
-        FFMPEGCommon.h
-        FFMPEGCommon.cpp
-        VideoDecoder.h
-        VideoDecoder.cpp
-        VideoDecoderOps.h
-        VideoDecoderOps.cpp
-        DeviceInterface.h
-    )
-    if(ENABLE_CUDA)
-        list(APPEND sources CudaDevice.cpp)
-    else()
-        list(APPEND sources CPUOnlyDevice.cpp)
-    endif()
-    add_library(${library_name} SHARED ${sources})
-    set_property(TARGET ${library_name} PROPERTY CXX_STANDARD 17)
+function(make_torchcodec_sublibrary
+    library_name
+    sources
+    dependent_libraries
+    ffmpeg_include_dirs)
```

Comment:
I could be wrong but it doesn't seem like we need to explicitly add `ffmpeg_include_dirs` anywhere. IIRC, the ffmpeg target we have are smart targets that will automatically add they INCLUDE dirs when calling `target_link_libraries`.

I tried removing those locally and things worked.

Reply:
Yup! Works for me too. I'm not sure what was going on when I added them. That will simplify the compilation as we don't need to pass as much info around.